### ISSUE TITLE
Nodes and edges not rendering properly after vite build

### ISSUE THREAD

**Issue by nartydev** [xyflow/xyflow#3879]
### Describe the Bug

Hello!

I have a reactflow which works correctly locally but when I deploy to a dev environment (i.e. once built) the edges and nodes are not displayed correctly...
I tried several things like statically setting the height and width of the parent, to see if it came from the ELK algorithm but still nothing..

### Steps to Reproduce the Bug or Issue

Here its the package I used:

```
"reactflow": "^11.10.3",
"react": "18.2.0",
"react-dom": "18.2.0",
"vite": "^4.2.1",
```

Maybe that have a link with vite build ? 

Here it's my code:

```
<ReactFlowProvider>
  <Box style={{ height: 620, width: '100%' }}>
      {initFitViewOptions && (
        <ReactFlow
          nodes={nodes}
          edges={edges}
          onNodesChange={onNodesChange}
          onEdgesChange={onEdgesChange}
          nodeTypes={nodeTypes}
          edgeTypes={edgeTypes}
          minZoom={-15}
          maxZoom={15}
          fitViewOptions={initFitViewOptions}
          fitView
          onNodeDragStop={(_, newNode) => onNodeDragStop(newNode)}
          onNodeClick={(_, node) => handleClickOnNode?.(node)}
        >
          <Controls showFitView />
        </ReactFlow>
      )}
      {!editorView && positionHaveChanged && (
        <HStack>
          <HStack>
            <Text color="greysecondary" mr="10px">
              You changed the positions. It won’t be visible by the other users.
            </Text>
            <Button
              variant="secondary"
              onClick={() => {
                handlePositionChanged?.(false);
                onLayout?.({ useInitialNodes: true });
              }}
            >
              Reset
            </Button>
          </HStack>
        </HStack>
      )}
    </Box>
<ReactFlowProvider>
```

### Expected behavior

I want that works after a build

### Screenshots or Videos

In local:
<img width="672" alt="Capture d’écran 2024-02-07 à 14 18 51" src="https://github.com/xyflow/xyflow/assets/16448382/c6d31e2e-864d-4555-869b-4158d2c11d03">

After a build:
<img width="684" alt="Capture d’écran 2024-02-07 à 14 19 29" src="https://github.com/xyflow/xyflow/assets/16448382/5be7250d-e5cf-45cb-864f-c2462af7aaad">

Width of nodes are not correct, edges doesn't appear, handler are positioned weirdly

### Platform

- OS: macOS
- Browser: Chrome
- Version: 111.0.5

### Additional context

Tell my if you want more informations
Thank in advance for help !

**Comment by moklick** [comment]
This looks weird. How do nodes get their dimensions? Are all styles loaded correctly? A repo would be super helpful to investigate further!

**Comment by moklick** [comment]
how does a node object look like when you put it in your nodes array?

**Comment by nartydev** [comment]
My nodes data look like this:
```
  {
      "id": "1",
      "type": "modelType",
      "data": {
        "entityName": "***",
        "tableName": "",
        "attributes": [
          {
            "attributeName": "***",
            "attributeType": "***",
            "keyType": "***"
          },
          {
            "attributeName": "***",
            "attributeType": "***",
            "keyType": ""
          }
        ]
      },
      "position": { "x": 0, "y": 0 },
      "targetPosition": "left",
      "sourcePosition": "right"
    },
    {
      "id": "22",
      "type": "modelType",
      "data": {
        "entityName": "***",
        "tableName": "",
        "attributes": [
          {
            "attributeName": "***",
            "attributeType": "***",
            "keyType": "***"
          },
          {
            "attributeName": "***",
            "attributeType": "***",
            "keyType": "***"
          }
        ]
      },
      "position": { "x": 0, "y": 0 },
      "targetPosition": "left",
      "sourcePosition": "right"
    }
```

**Comment by nartydev** [comment]
I will try to explicitly define width for nodes for check if that change something

**Comment by nartydev** [comment]
> If you want to set width/height, you would need to do it via `node.style: { width: 100, height 100 }`.


<img width="685" alt="Capture d’écran 2024-02-08 à 14 20 11" src="https://github.com/xyflow/xyflow/assets/16448382/34bc3256-6fbe-4fd9-8ef8-31480ec4d293">

Ok, so I try to set width that display correctly the nodes but the edges nop... same for the position of the handler that positioned with absolute value..

**Comment by moklick** [comment]
what is happening here 😳 do you get any warnings about the edges in the console? what is going on with the handle positions? You should be able to inspect them in order to see why they are positioned that way.

**Comment by nartydev** [comment]
<img width="676" alt="Capture d’écran 2024-02-08 à 14 56 53" src="https://github.com/xyflow/xyflow/assets/16448382/ff157896-e36f-46a2-acc6-df0d119f0d21">

You are right I I succeeded to display right the handlers
But for edges I have nothing in my console (I think it's because it's a production build, I have try the flag --mode development in vite build but nothing appear...), in dev all my edges are linked correctly so I don't know..

**Comment by moklick** [comment]
and there are no edges in the DOM?

**Comment by nartydev** [comment]
Dev version:
<img width="1512" alt="Capture d’écran 2024-02-08 à 15 12 47" src="https://github.com/xyflow/xyflow/assets/16448382/babe41f1-3fe7-4c32-8839-c8a585d8f35f">

Build version:
<img width="1313" alt="Capture d’écran 2024-02-08 à 15 12 08" src="https://github.com/xyflow/xyflow/assets/16448382/9923ca01-4d34-4d35-9bbc-d32e2044cd76">
All edges are here I can see it in DOM

**Comment by moklick** [comment]
can you try to change the stroke color manually?

**Comment by nartydev** [comment]
![Uploading Capture d’écran 2024-02-08 à 15.37.31.png…]()
Yep, nothing change

**Comment by nartydev** [comment]
<img width="1308" alt="Capture d’écran 2024-02-08 à 15 45 41" src="https://github.com/xyflow/xyflow/assets/16448382/5c0bbbeb-2e82-4e33-83e1-dc2e513552ba">
In dev

but in build we doesn't have this attribute set, I try to add it manually in DOM but nothing change..
The zoom are weird also 
[screen-capture.webm](https://github.com/xyflow/xyflow/assets/16448382/8345e1db-d55f-41cc-897d-665182a31ee5)

**Comment by moklick** [comment]
can you show the styles of the SVG with the class "react-flow__edges" in prod?

**Comment by nartydev** [comment]
<img width="1361" alt="Capture d’écran 2024-02-08 à 16 15 27" src="https://github.com/xyflow/xyflow/assets/16448382/62f5448d-3749-437e-8f12-04a330d0ad11">
Doesn't set

**Comment by moklick** [comment]
That explains it. The reactflow styles are not there.

**Comment by nartydev** [comment]
You are right, it came from the config quickly thank you very much for the help!
